src/menu.js

```diff
@@ -116,28 +125,28 @@ class Menu extends Component {
   componentWillUnmount() {
     this.autocomplete.removeMenu(this)
     this.autocomplete.emitter.off('menu:open', this.setDefaultHighlightedIndex)
+
+    // cleanup MenuStatus component
+    ReactDOM.unmountComponentAtNode(this._statusNode)
+    document.body.removeChild(this._statusNode)
```

Comment:
Good call :+1:

I wonder if we could move all this `componentWillUnmount` stuff to a method called `cleanup` that's set on the instance in `componentDidMount` that way `componentWillUnmount` just calls `cleanup` and we keep this logic closer together?